src/scanner.h

```diff
@@ -141,8 +141,8 @@ class Scanner {
   struct SimpleKey {
     SimpleKey(const Mark &mark_, std::size_t flowLevel_);
 
-    void Validate();
-    void Invalidate();
+    void Validate() const;
```

Comment:
Although the implementation for these is technically const, it's misleading, since the implementations modify underlying state; just leave them off.